tests/capture/test_nested_plxpr.py

```diff
@@ -159,3 +161,111 @@ def qfunc(wire):  # x is closure variable and a tracer
 
         assert len(q) == 1
         qml.assert_equal(q.queue[0], qml.adjoint(qml.RX(2.5, 2)))
+
+
+class TestCtrlQfunc:
+    """Tests for the ctrl primitive."""
+
+    def test_operator_type_input(self):
+        """Test that an operator type can be the callable."""
+
+        def f(x, w):
+            return qml.ctrl(qml.RX, 1)(x, w)
+
+        plxpr = jax.make_jaxpr(f)(0.5, 0)
+
+        with qml.queuing.AnnotatedQueue() as q:
+            out = jax.core.eval_jaxpr(plxpr.jaxpr, plxpr.consts, 1.2, 2)
+
+        assert f(0.5, 0) is None
+        assert out == []
+        expected = qml.ctrl(qml.RX(1.2, 2), 1)
+        qml.assert_equal(q.queue[0], expected)
+
+        assert plxpr.eqns[0].primitive == ctrl_prim
+        assert plxpr.eqns[0].params["control_values"] == [True]
+        assert plxpr.eqns[0].params["n_control"] == 1
+        assert plxpr.eqns[0].params["work_wires"] is None
+
+    def test_dynamic_control_wires(self):
+        """Test that control wires can be dynamic."""
+
+        def f(w1, w2, w3):
+            return qml.ctrl(qml.X, (w2, w3))(w1)
+
+        plxpr = jax.make_jaxpr(f)(4, 5, 6)
+
+        with qml.queuing.AnnotatedQueue() as q:
+            out = jax.core.eval_jaxpr(plxpr.jaxpr, plxpr.consts, 1, 2, 3)
+
+        assert out == []
+        expected = qml.Toffoli(wires=(2, 3, 1))
+        qml.assert_equal(q.queue[0], expected)
+        assert len(q) == 1
+
+        assert plxpr.eqns[0].primitive == ctrl_prim
+        assert plxpr.eqns[0].params["control_values"] == [True, True]
+        assert plxpr.eqns[0].params["n_control"] == 2
+        assert plxpr.eqns[0].params["work_wires"] is None
+
+    def test_work_wires(self):
+        """Test that work wires can be provided."""
+
+        def f(w):
+            return qml.ctrl(qml.S, (1, 2), work_wires="aux")(w)
+
+        plxpr = jax.make_jaxpr(f)(6)
+
+        with qml.queuing.AnnotatedQueue() as q:
+            out = jax.core.eval_jaxpr(plxpr.jaxpr, plxpr.consts, 5)
+
+        assert out == []
+        expected = qml.ctrl(qml.S(5), (1, 2), work_wires="aux")
+        qml.assert_equal(q.queue[0], expected)
+        assert len(q) == 1
+
+        assert plxpr.eqns[0].params["work_wires"] == "aux"
+
+    def test_control_values(self):
+        """Test that control values can be provided."""
+
+        def f(z):
+            return qml.ctrl(qml.RZ, (3, 4), [False, True])(z, 0)
+
+        plxpr = jax.make_jaxpr(f)(0.5)
+
+        with qml.queuing.AnnotatedQueue() as q:
+            out = jax.core.eval_jaxpr(plxpr.jaxpr, plxpr.consts, 5.4)
+
+        assert out == []
+        expected = qml.ctrl(qml.RZ(5.4, 0), (3, 4), [False, True])
+        qml.assert_equal(q.queue[0], expected)
+        assert len(q) == 1
+
+        assert plxpr.eqns[0].params["control_values"] == [False, True]
+        assert plxpr.eqns[0].params["n_control"] == 2
+
+    def test_nested_control(self):
```

Comment:
Do we also need a test case where we pass a callable to `ctrl` that produces multiple operations?

Reply:
adding.